File: packages/cli/src/commands/restart.ts
import { getConfig } from '../utils/config.js';
import { stopCommand } from './stop.js';
import { startCommand } from './start.js';

export type RestartOptions = {
  native?: boolean;
};

export async function restartCommand(options: RestartOptions = {}): Promise<void> {
  const { port } = getConfig();
  if (options.native) {
    await stopCommand({ native: true });
    await startCommand({ port: String(port), detach: true, native: true });
    return;
  }

  await stopCommand({ remove: true });
  await startCommand({ port: String(port), detach: true });
}

File: packages/cli_native/src/commands/restart.rs
use anyhow::Result;

use crate::{
    commands::{
        start::{start_command, StartOptions},
        stop::stop_command,
    },
    utils::config::get_config,
};

pub async fn restart_command() -> Result<()> {
    let port = get_config().port;
    stop_command(true).await?;
    start_command(StartOptions {
        port: Some(port),
        detach: true,
        no_update_check: false,
        gpu: true,
        audio: true,
        display: true,
        input: true,
    })
    .await
}
